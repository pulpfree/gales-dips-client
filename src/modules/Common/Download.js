import React, { Component } from 'react'
import PropTypes from 'prop-types'

import AppBar from '@material-ui/core/AppBar'
import Button from '@material-ui/core/Button'
import Close from '@material-ui/icons/Close'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'

class Download extends Component {
  handleCloseWindow = () => {
    window.close()
  }

  render() {
    const { classes } = this.props

    return (
      <div>
        <AppBar
          className={classes.appBar}
          color="secondary"
          position="static"
        >
          <Typography
            className={classes.title}
            gutterBottom
            onClick={this.handleCloseWindow}
            variant="h6"
          >Gales Dips - File Download
          </Typography>
        </AppBar>
        <div className={classes.container}>
          <Typography
            gutterBottom
            variant="h6"
          >File Download
          </Typography>
          <div className={classes.message}>
            Stand by your file is being prepared and will download shortly.
          </div>
          <Button
            className={classes.button}
            color="secondary"
            onClick={this.handleCloseWindow}
          >
            <Close className={classes.leftIcon} />
          Close Window
          </Button>
        </div>
      </div>
    )
  }
}
Download.propTypes = {
  classes: PropTypes.instanceOf(Object).isRequired,
}

const styles = theme => ({
  appBar: {
  },
  button: {
    width: 180,
  },
  container: {
    display: 'flex',
    flexDirection: 'column',
    fontFamily: theme.typography.fontFamily,
    margin: 'auto',
    marginTop: theme.spacing.unit * 10,
    width: 500,
  },
  leftIcon: {
    marginRight: theme.spacing.unit,
  },
  message: {
    marginBottom: theme.spacing.unit * 3,
    marginTop: theme.spacing.unit * 2,
  },
  title: {
    color: '#fff',
    padding: theme.spacing.unit * 2,
    cursor: 'pointer',
  },
})

export default withStyles(styles)(Download)
