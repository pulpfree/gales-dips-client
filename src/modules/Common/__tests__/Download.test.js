/* eslint no-undef: "off" */

import React from 'react'
import { shallow } from 'enzyme'

import Download from '../Download'

describe('Download', () => {
  let mountedC
  let props
  const reportC = () => {
    if (!mountedC) {
      mountedC = shallow(<Download {...props} />)
    }
    return mountedC
  }

  beforeEach(() => {
    props = {
    }
    mountedC = undefined
  })

  it('matches snapshot', () => {
    expect(reportC()).toMatchSnapshot()
  })
})
