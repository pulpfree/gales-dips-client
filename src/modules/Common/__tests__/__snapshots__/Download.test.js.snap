// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Download matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Download) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "appBar": "Download-appBar-1",
        "button": "Download-button-2",
        "container": "Download-container-3",
        "leftIcon": "Download-leftIcon-4",
        "message": "Download-message-5",
        "title": "Download-title-6",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "appBar": "Download-appBar-1",
          "button": "Download-button-2",
          "container": "Download-container-3",
          "leftIcon": "Download-leftIcon-4",
          "message": "Download-message-5",
          "title": "Download-title-6",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
